app/modules/seed_library/endpoints_seed_library.py

```diff
@@ -0,0 +1,506 @@
+    """
+
+    if not (
+        is_user_member_of_any_group(
+            user=user,
+            allowed_groups=[GroupType.seed_library],
+        )
+    ):
+        raise HTTPException(
+            status_code=403,
+            detail="You are not allowed to see plants from this user",
+        )
+    return await cruds_seed_library.get_plants_by_user_id(user_id, db)
+
+
+@module.router.get(
+    "/seed_library/plants/users/me",
+    response_model=list[schemas_seed_library.PlantSimple],
+    status_code=200,
+)
+async def get_my_plants(
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user()),
+):
+    """
+    Return all plants where user ={user_id} from database as a list of PlantsComplete schemas
+    """
+    return await cruds_seed_library.get_plants_by_user_id(user.id, db)
+
+
+@module.router.get(
+    "/seed_library/plants/{plant_id}",
+    response_model=schemas_seed_library.PlantSimple,
+    status_code=200,
+)
+async def get_plant_by_id(
+    plant_id: uuid.UUID,
+    db: AsyncSession = Depends(get_db),
+):
+    """
+    Return the plants where plant ={plant_id} from database as a PlantsComplete schemas
+    """
+    plant = await cruds_seed_library.get_plant_by_id(plant_id, db)
+    if plant is None:
+        raise HTTPException(404, "Plant does not exist.")
+    return plant
+
+
+@module.router.post(
+    "/seed_library/plants/",
+    response_model=schemas_seed_library.PlantComplete,
+    status_code=201,
+)
+async def create_plant(
+    plant_base: schemas_seed_library.PlantCreation,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user()),
+):
+    """
+    Create a new Plant by giving an PlantCreation scheme
+    **This endpoint is only usable if the plant has an ancestor_id or by seed_library **
+    """
+    if plant_base.ancestor_id is None:
+        if not is_user_member_of_any_group(
+            user=user,
+            allowed_groups=[GroupType.seed_library],
+        ):
+            raise HTTPException(
+                status_code=403,
+                detail="Please contact an Planet&co member if you are trying to donate a plant that does not come from the seed library",
+            )
+
+    species_reference = await cruds_seed_library.get_species_by_id(
+        plant_base.species_id,
+        db,
+    )
+    if species_reference is None:
+        raise HTTPException(
+            404,
+            "Species not found",
+        )
+    date = datetime.now(tz=UTC)
+    if species_reference:
+        plant_reference = (
+            str(species_reference.prefix)
+            + str(date.day)
+            + str(date.month)
+            + str(date.year)
+        )
+        plant_number = await cruds_seed_library.count_plants_created_today(
+            plant_reference,
+            db,
+        )
+
+        plant_reference = (
+            plant_reference + "0" * (3 - len(str(plant_number))) + str(plant_number)
+        )
```

Comment:
You can use `{var:03}` to pad the integer variable `var` with 0:
```suggestion
        plant_reference = f"{species_reference.prefix}{date.day}{date.month}{date.year}{plant_reference}{plant_number:03}"
```